src/layer/vector/Renderer.js

```diff
@@ -15,7 +15,20 @@ L.Renderer.include({
      * @listens L.Map~rotate
      */
     getEvents: function() {
-        return L.extend(rendererProto.getEvents.apply(this, arguments), { rotate: this._update })
+        return L.extend(rendererProto.getEvents.apply(this, arguments), { rotate: this._update });
+    },
+
+    /**
+     * Fix for `map.flyTo()` when `false === map.options.zoomAnimation`
+     * 
+     * @see https://github.com/Leaflet/Leaflet/pull/8794
+     */
+    onAdd: function() {
+        rendererProto.onAdd.apply(this, arguments);
+        if (L.version <= "1.9.3") {
+            // always keep transform-origin as 0 0
+            this._container.classList.add('leaflet-zoom-animated');
+        }
```

Comment:
Ref: https://github.com/Raruto/leaflet-rotate/pull/22/commits/2de9c8aabf79aeb23ceef380506ad0b870f8fb5f and https://github.com/Raruto/leaflet-rotate/pull/22#pullrequestreview-1283232182

@heyajohnny Did you mean to apply this patch?

Unfortunately the problem reported in here is not solved with this patch.

https://github.com/Raruto/leaflet-rotate/blob/8e9f36858e8172931972b3adf68b23dfc74527fd/src/layer/vector/Renderer.js#L21-L55

To check this more clearly you can always set `zoomAnimation: false` and toggle between zoom levels between 0 and 2.

During the zoom animation `L.Renderer::_updateTransform(zoom, center)` breaks the calculation of the origin of the panes (it is less noticeable with `zoomAnimation: true` because at the end of the zoom animation `L.Renderer::_update()` is called last which repositions correctly all coordinates by scaling them to the current zoom level, i.e. going from css transform `scale(0.5)` to `scale(1)`)

if you're interested in finding a solution I can publish a branch that where I'm doing some visual debugging on (although it's a bit messed up 😄).

Currently the problem occurs especially between zoom levels 0 and 2, due to the use of the following function to calculate the offset between the current origin (eg. at `zoomStart = 1`) and the next one (eg. at `zoomEnd = 0`).

```js
offset = this._map._latLngToNewLayerPoint(this._topLeft, zoom, center);
```

Basically, a wrong approximation of the origin is made because in that point the screen coordinates (pixel) are projected / converted into geographical coordinates (latlng) and vice-versa. And at that moment the origin is "shifted" from the real position (because there is no lat/lng associated for the current origin location which is outside the map bounds).

Perhaps an optimal solution would be to follow the calculation done by the core (which instead starts on the center of the container to deduce the offset of the origin pixel), but for the moment this doesn't work because it would be necessary to update these calculations to pass into the "rotated" reference system (applying the relative rotation functions to the various pixel coordinates)

```js
// Leaflet/Leaflet/src/layer/vector/Renderer.js#L84-L93

_updateTransform(center, zoom) {
  const scale = this._map.getZoomScale(zoom, this._zoom),
    viewHalf = this._map.getSize().multiplyBy(0.5 + this.options.padding),
    currentCenterPoint = this._map.project(this._center, zoom),

    topLeftOffset = viewHalf.multiplyBy(-scale).add(currentCenterPoint)
				  .subtract(this._map._getNewPixelOrigin(center, zoom));

    DomUtil.setTransform(this._container, topLeftOffset, scale);
},
```

Ref: [Leaflet/Leaflet/src/layer/vector/Renderer.js#L84-L93](https://github.com/Leaflet/Leaflet/blob/d1016279f37652a78b53c35e826b3180e4f7f03f/src/layer/vector/Renderer.js#L84-L93)